src/test/shared/utilities/editorUtilities.test.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as assert from 'assert'
+import * as path from 'path'
+import * as semver from 'semver'
+import { closeAllEditors, assertTabCount, createTestWorkspaceFolder, openATextEditorWithText } from '../../testUtil'
+import { getOpenFilesInWindow } from '../../../shared/utilities/editorUtilities'
+
+// VSCode tab APIs are available since 1.68.0
+function shouldRunTheTest(): boolean {
+    return (semver.valid(vscode.version) && semver.gte(vscode.version, '1.68.0')) as boolean
+}
+
+describe('supplementalContextUtil', function () {
+    let tempFolder: string
+
+    describe('getOpenFilesInWindow', async function () {
+        before(async function () {
+            this.timeout(60000)
+        })
+
+        beforeEach(async function () {
+            tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+        })
+
+        afterEach(async function () {
+            await closeAllEditors()
```

Comment:
I'd move this to the `beforeEach` to ensure there are no open editors before each test runs, a completely separate test may have opened an editor and the first test would get the wrong result.

Reply:
@nkomonen-amazon @justinmk3 should we keep this cleanup `closeAllEditors` both on `beforeEach` and `afterEach`?

Reply:
asking because I saw it's used in `after` in `visualizeStateMachine.test.ts`
https://github.com/aws/aws-toolkit-vscode/blob/bfd220fb09749ef345fb94c58fabb668ae9fa5b3/src/testInteg/stepFunctions/visualizeStateMachine.test.ts#L116

Reply:
not sure if it will cause unclean states if it's removed from `afterEach`

Reply:
> should we keep this cleanup `closeAllEditors` both on `beforeEach` and `afterEach`?

sure

Reply:
I would say in general we should clean the state in the `before` to make tests less fragile from external changes. And doing it in the `after` is a courtesy for other tests. But each test in general shouldn't rely on a clean state when first starting.

I'd put `closeAllEditors()` in the `beforeEach` for your tests, and `after` as the courtesy for subsequent tests. If tests are failing after this change then we'll need to fix those tests.

Reply:
gotcha, yea added on both sites